doc/types.md

```diff
@@ -0,0 +1,59 @@
+# Types
+
+Every programming language needs a foundation in sound type theory.
+Lean has a rich provable type theory that is built on type universes.
+
+TODO: add a profound insight into why type universes are useful -

```

Comment:
As far as I understand, type universes are mostly there to avoid self-referential paradoxes like Russell's paradox.